src/js/angular/utils/hash-utils.js

```diff
@@ -0,0 +1,6 @@
+import {KJUR} from "jsrsasign";
+
+export const generateMD5Hash = (str) => {
+    const md = new KJUR.crypto.MessageDigest({alg: 'md5', prov: 'cryptojs'});
```

Comment:
Can this be created once, instead of creating it for each resource?

Reply:
yeah, sure. I thought it might not be needed in this case, but I'll change it.